pokemon.js

```diff
@@ -0,0 +1,75 @@
+let section =document.getElementById("container");
+let whencl=document.getElementById("whencl");
+//for searchbar
+const searchInput = document.querySelector('.input');
+let arr=[];
+searchInput.addEventListener('input',(input)=>{
+    let value=input.target.value;
+    const filteredEle=arr.filter(ele => {
+        let isVisible=ele.name.includes(value);
+       return isVisible;
+    });
+    display(filteredEle);
+});
+function hoverfun(para){
+    para.style.position="relative";
+   para.style.animation="mynewmove .2s 2";
+}
+function closeOverlay(){
+    whencl.style.zIndex=-1;
+    whencl.style.opacity='0';
+    whencl.innerHTML='';
+}
+//when clicking on any pokemon
+async function clickfun(url,imgn){
+    let response= await fetch(url);
+    let data= await response.json();
+    let types=data.types;
+    let type='';
+    for(let i=0;i<types.length;i++){
```

Comment:
let's use const for read inly variables, and please for all local variables, use a descriptive name. for example pokemonType instead of type.